src/Development/IDE/Spans/Calculate.hs

```diff
@@ -107,31 +109,44 @@ ieLNames _ = []
 
 -- | Get the name and type of a binding.
 getTypeLHsBind :: (GhcMonad m)
-               => OccEnv (HsBind GhcPs)
+               => [TypecheckedModule]
+               -> OccEnv (HsBind GhcPs)
                -> LHsBind GhcTc
-               -> m [(SpanSource, SrcSpan, Maybe Type)]
-getTypeLHsBind funBinds (L _spn FunBind{fun_id = pid})
-  | Just FunBind {fun_matches = MG{mg_alts=L _ matches}} <- lookupOccEnv funBinds (occName $ unLoc pid) =
-  return [(Named (getName (unLoc pid)), getLoc mc_fun, Just (varType (unLoc pid))) | match <- matches, FunRhs{mc_fun = mc_fun} <- [m_ctxt $ unLoc match] ]
+               -> m [(SpanSource, SrcSpan, Maybe Type, [T.Text])]
+getTypeLHsBind tms funBinds (L _spn FunBind{fun_id = pid})
+  | Just FunBind {fun_matches = MG{mg_alts=L _ matches}} <- lookupOccEnv funBinds (occName $ unLoc pid) = do
+  let name = getName (unLoc pid)
+  docs <- getDocumentationTryGhc' tms name
+  return [(Named name, getLoc mc_fun, Just (varType (unLoc pid)), docs) | match <- matches, FunRhs{mc_fun = mc_fun} <- [m_ctxt $ unLoc match] ]
 -- In theory this shouldn’t ever fail but if it does, we can at least show the first clause.
-getTypeLHsBind _ (L _spn FunBind{fun_id = pid,fun_matches = MG{}}) =
-  return [(Named $ getName (unLoc pid), getLoc pid, Just (varType (unLoc pid)))]
-getTypeLHsBind _ _ = return []
+getTypeLHsBind tms _ (L _spn FunBind{fun_id = pid,fun_matches = MG{}}) = do
+  let name = getName (unLoc pid)
+  docs <- getDocumentationTryGhc' tms name
+  return [(Named name, getLoc pid, Just (varType (unLoc pid)), docs)]
+getTypeLHsBind _ _ _ = return []
 
 -- | Get the name and type of an expression.
 getTypeLHsExpr :: (GhcMonad m)
-               => TypecheckedModule
+               => [TypecheckedModule]
                -> LHsExpr GhcTc
-               -> m (Maybe (SpanSource, SrcSpan, Maybe Type))
-getTypeLHsExpr _ e = do
+               -> m (Maybe (SpanSource, SrcSpan, Maybe Type, [T.Text]))
+getTypeLHsExpr tms e = do
   hs_env <- getSession
   (_, mbe) <- liftIO (deSugarExpr hs_env e)
-  return $
-    case mbe of
-      Just expr ->
-        Just (getSpanSource (unLoc e), getLoc e, Just (CoreUtils.exprType expr))
-      Nothing -> Nothing
+  case mbe of
+    Just expr -> do
+      let ss = getSpanSource' e
+      docs <- case ss of
+                Named n -> getDocumentationTryGhc' tms n
+                _       -> return []
+      return $ Just (ss, getLoc e, Just (CoreUtils.exprType expr), docs)
+    Nothing -> return Nothing
   where
+    getSpanSource' :: LHsExpr GhcTc -> SpanSource
```

Comment:
I think this code would be easier to read if you merge `getSpanSource'` and `getSpanSource`.